libs/encryption-sdk/src/main/java/org/opensearch/encryption/frame/core/FrameDecryptionHandler.java

```diff
@@ -0,0 +1,296 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.encryption.frame.core;
+
+import com.amazonaws.encryptionsdk.CryptoAlgorithm;
+import com.amazonaws.encryptionsdk.exception.AwsCryptoException;
+import com.amazonaws.encryptionsdk.exception.BadCiphertextException;
+import com.amazonaws.encryptionsdk.internal.CryptoHandler;
+import com.amazonaws.encryptionsdk.internal.ProcessingSummary;
+import com.amazonaws.encryptionsdk.model.CipherFrameHeaders;
+
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import java.util.Arrays;
+
+/**
+ * This class implementation has been referenced from AWS encryption SDK.
+ * https://tiny.amazon.com/1j988pk3f/codeamazpackblobheadamaz4
```

Comment:
nit: this might not be publicly accessible?

Reply:
Ya, I will correct these.